zebra-chain/src/transaction/tests/vectors.rs

```diff
@@ -341,3 +363,87 @@ fn invalid_orchard_nullifier() {
         SerializationError::Parse("Invalid pallas::Base value for orchard Nullifier").to_string()
     );
 }
+
+/// Do a round-trip test via librustzcash on fake v5 transactions created from v4 transactions
+/// in the block test vectors.
+/// Makes sure that zebra-serialized transactions can be deserialized by librustzcash.
+#[test]
+fn fake_v5_librustzcash_round_trip() {
+    zebra_test::init();
+
+    fake_v5_librustzcash_round_trip_for_network(Network::Mainnet);
+    fake_v5_librustzcash_round_trip_for_network(Network::Testnet);
+}
+
+fn fake_v5_librustzcash_round_trip_for_network(network: Network) {
+    let block_iter = match network {
+        Network::Mainnet => zebra_test::vectors::MAINNET_BLOCKS.iter(),
+        Network::Testnet => zebra_test::vectors::TESTNET_BLOCKS.iter(),
+    };
+
+    for (height, original_bytes) in block_iter {
+        let original_block = original_bytes
+            .zcash_deserialize_into::<Block>()
+            .expect("block is structurally valid");
+
+        // skip blocks that are before overwinter as they will not have a valid consensus branch id
+        if *height
+            < NetworkUpgrade::Overwinter
+                .activation_height(network)
+                .expect("a valid height")
+                .0
+        {
+            continue;
+        }
```

Comment:
Suggestion: The way this is currently written deserializes every  block, even those that are ignored. It might make sense to reorder the height check, but maybe it's even better to filter those out sooner:

```suggestion
    let block_iter = match network {
        Network::Mainnet => zebra_test::vectors::MAINNET_BLOCKS.iter(),
        Network::Testnet => zebra_test::vectors::TESTNET_BLOCKS.iter(),
    };

    let overwinter_activation_height = NetworkUpgrade
        .activation_height(network)
        .expect("a valid height")
        .0;

    // skip blocks that are before overwinter as they will not have a valid consensus branch id
    let blocks_after_overwinter = blocks_iter.skip_while(|(height, _)| {
        *height < overwinter_activation_height
    });

    for (height, original_bytes) in blocks_after_overwinter {
        let original_block = original_bytes
            .zcash_deserialize_into::<Block>()
            .expect("block is structurally valid");
```

I used `skip_while` because the block test vectors are ordered by height, but it might make sense to use `filter` instead to be more defensive in case it's not ordered in the future.

Reply:
The ordering is a requirement of `CONTINUOUS_{MAIN,TEST}NET_BLOCKS`, so it's unlikely to change.

Reply:
Good idea, added in 29b13a0. I've also changed a previous test which this test was based on.